pkg/util/platformworkloadidentity/federatedcredentials.go

```diff
@@ -18,16 +20,17 @@ const (
 )
 
 func GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, identityResourceId azure.Resource, serviceAccountName string) string {
-	clusterResourceKey := fmt.Sprintf("%s-%s-%s", clusterResourceId.SubscriptionID, clusterResourceId.ResourceGroup, clusterResourceId.ResourceName)
-	name := fmt.Sprintf("%s-%s-%s", clusterResourceKey, serviceAccountName, identityResourceId.ResourceName)
+	sanitizedServiceAccountName := strings.Join((strings.Split(strings.ReplaceAll(serviceAccountName, ":", "-"), "-"))[2:], "-")
```

Comment:
Consider using strings.TrimPrefix() instead, since the service account name will always start with `system:serviceaccount:`.

Consider also dropping the namespace from the federated credential name. You could do this by splitting the service account name string like you do now, but keep only the last element.